src/main/common/parse.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+
+#include "common/utils.h"
+#include "utils.h"
+
+enum argType_e {
+    ARG_TYPE_SEP,
+    ARG_TYPE_STRING,
+    ARG_TYPE_STRING_VARIANT,
+    ARG_TYPE_INT,
+    ARG_TYPE_INT_IN_RANGE,
+    ARG_TYPE_INT_CONVERT,
+};
+
+typedef struct convertResult_s {
+    bool isOk;
+    union {
+        long int value;
+        const char *errMsg;
+    };
+} convertResult_t;
+
+typedef convertResult_t convertFn(long int value);
+
+enum storeType_e {
+    ARG_STORE_NULL,
```

Comment:
NULL is `(void*)0` in C
```suggestion
    ARG_STORE_NIL,
```